scanner/src/main/java/edu/ucr/cs/riple/scanner/ErrorProneCLIFlagsConfig.java

```diff
@@ -86,7 +80,7 @@ public ErrorProneCLIFlagsConfig(ErrorProneFlags flags) {
     this.classTrackerIsActive =
         XMLUtil.getValueFromAttribute(document, "/scanner/class", "active", Boolean.class)
             .orElse(false);
-    this.serializer = (this.outputDirectory == null) ? null : new Serializer(this);
+    this.serializer = new Serializer(this);
```

Comment:
I see that you removed the null check for `this.outputdirectory` but i cant see the check moved inside the serializer constructor